src/py/mat3ra/made/tools/analyze.py

```diff
@@ -280,17 +284,26 @@ def get_nearest_neighbors_atom_indices(
         coordinate = [0, 0, 0]
     structure = to_pymatgen(material)
     voronoi_nn = PymatgenVoronoiNN(
-        tol=0.5,
-        cutoff=15.0,
-        allow_pathological=False,
+        tol=tolerance,
+        cutoff=cutoff,
         weight="solid_angle",
-        extra_nn_info=True,
+        extra_nn_info=False,
         compute_adj_neighbors=True,
     )
-    structure.append("X", coordinate, validate_proximity=False)
-    neighbors = voronoi_nn.get_nn_info(structure, len(structure.sites) - 1)
+    coordinates = material.basis.coordinates
+    coordinates.filter_by_values([coordinate])
```

Comment:
This seems very inefficient.

We should have an option to pass site index directly and get the coordinate from it.

the corresponding logic can also go to `get_site_index_from_coordinate`

Reply:
Ah, yes. I added in the next branch the get_id_by_value. will add here